### DISCUSSION TITLE
Nuget - pin dependencies

### DISCUSSION THREAD

**Discussion by mieliespoor** [renovatebot/renovate#20432] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

32.240.5

### If you're self-hosting Renovate, select which platform you are using.

Bitbucket Server

### If you're self-hosting Renovate, tell us what version of the platform you run.

7.21

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

Before pin:

```
<PackageReference Include="AWSSDK.S3" Version="3.5.*" />
<PackageReference Include="Microsoft.AspNetCore.Mvc.Versioning.ApiExplorer" Version="5.0.*" />
```

After pin
```
<PackageReference Include="AWSSDK.S3" Version="3.5.8.5" />
<PackageReference Include="Microsoft.AspNetCore.Mvc.Versioning.ApiExplorer" Version="5.0.0" />
```

Renovate makes use of dependencies pinning for npm, but we don't see the same for nuget. Nuget has various ways in which versions can be specified. I haven't however seen renovate pin the dependency versions. This is a problem as it take a lot work work to find these and it negates the impact that renovate does have.

### What you tried so far.

Can't get it to work as I don't think Renovate supports this?

### Relevant debug logs

<details><summary>Logs</summary>

```
Copy/paste the relevant log(s) here, between the starting and ending backticks
```

</details>

**Comment by viceice** [comment, marked as answer]
not yet supported, nuget versioning needs an update for it

**Reply by mieliespoor** [reply]
Not sure what you mean by nuget versioning needs an update? If a package uses semver, what can or should change?

**Reply by mieliespoor** [reply]
Can renovate not just pick the next biggest patch version in the example I showed? Surely that could work?

**Reply by rarkins** [reply]
PRs welcome

**Reply by viceice** [reply]
this class needs to be changed

https://github.com/renovatebot/renovate/blob/e41681db3d2b4fc2ee47eaa2c6ad5b6a5d82b189/lib/modules/versioning/nuget/index.ts#L15
